<launch>

    <!-- Velocity smoother -->
    <!--
    <include file="$(find kaqi_teleop)/launch/velocity_smoother.launch.xml" />
    -->

    <!--  smooths inputs from cmd_vel_mux/input/teleop_raw to cmd_vel_mux/input/teleop -->
    <node pkg="nodelet" type="nodelet" name="teleop_velocity_smoother" args="load yocs_velocity_smoother/VelocitySmootherNodelet /mobile_base_nodelet_manager">
        <param name="speed_lim_v" value="0.8" />
        <param name="speed_lim_w" value="1.2" />
        <param name="accel_lim_v" value="1.0" />
        <param name="accel_lim_w" value="2.0" />
        <param name="frequency" value="20" />
        <param name="decel_factor" value="1.6" />
        <param name="robot_feedback" value="0" />

        <remap from="teleop_velocity_smoother/raw_cmd_vel" to="teleop/raw_cmd_vel"/>
        <remap from="teleop_velocity_smoother/smooth_cmd_vel" to="base_controller/command"/>
        <!-- Robot velocity feedbacks; use the one configured as base default -->
        <remap from="teleop_velocity_smoother/odometry" to="/odom"/>
        <remap from="teleop_velocity_smoother/robot_cmd_vel" to="base_controller/command"/>
    </node>

    <node pkg="kaqi_teleop" type="teleop_kaqi_keyboard" name="teleop_kaqi_keyboard" args="--deadman_no_publish" output="screen" respawn="true">
        <!-- Remap topic -->
        <remap from="cmd_vel" to="teleop/raw_cmd_vel" />

        <!-- Velocity Limit -->
        <param name="walk_vel" value="0.3" type="double"/>
        <param name="run_vel" value="0.6" type="double"/>
        <param name="yaw_rate" value="0.6" type="double"/>
        <param name="yaw_run_rate" value="1.2" type="double"/>

    </node>

</launch>
